dd-java-agent/instrumentation/play-2.6/src/main/java8/datadog/trace/instrumentation/play26/PlayHttpServerDecorator.java

```diff
@@ -92,4 +93,30 @@ public AgentSpan onError(final AgentSpan span, Throwable throwable) {
     }
     return super.onError(span, throwable);
   }
+
+  private Option<HandlerDef> getDefOption(TypedKey handler, Request request) {
+    Class HandlerDefClass = handler.getClass();
+    Method asScalaOrUnderlying = null;
+
+    try {
+      asScalaOrUnderlying = HandlerDefClass.getMethod("asScala");
+    } catch (NoSuchMethodException e) {
+      try {
+        asScalaOrUnderlying = HandlerDefClass.getMethod("underlying");
+      } catch (NoSuchMethodException err) {
+        log.error(err.getMessage());
+        return null;
+      }
+    }
+
+    Option<HandlerDef> defOption = null;
+    try {
+      defOption = request.attrs().get((play.api.libs.typedmap.TypedKey) asScalaOrUnderlying.invoke(Router.Attrs.HANDLER_DEF));
```

Comment:
It's preferable to check before casting:
```suggestion
      Object typedKey = null;
      try {
        typedKey = asScalaOrUnderlying.invoke(Router.Attrs.HANDLER_DEF);
      } catch (ReflectiveOperationException e) {
        log.error(e.getMessage());
      }
      if (typedKey instanceof play.api.libs.typedmap.TypedKey) {
        return request.attrs().get((play.api.libs.typedmap.TypedKey) typedKey);
      } else {
        log.error("Unexpected type for TypedMap key: " + typedKey);
      }
      return null;
```

I've also opted to use the base exception type.